compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -478,9 +478,19 @@ impl<'a> Resolver<'a> {
                         name
                     ),
                 );
-                err.help(
-                    &format!("it is currently only allowed to use either `{0}` or `{{ {0} }}` as generic constants", name)
-                );
+
+                if is_type {
+                    err.note(&format!(
+                        "type parameters are currently not permitted in anonymous constants"
+                    ));
```

Comment:
```suggestion
                    err.note("type parameters are currently not permitted in anonymous constants");
```

Reply:
I had it that way originally, I ran `rustfmt` and this popped out. Would you rather I changed it back?

Reply:
It's not about changing the formatting here. The relevant part is the removal of `&format!`.